internal/test/scorecard/tests.go

```diff
@@ -146,24 +146,23 @@ func CryostatConfigChangeTest(bundle *apimanifests.Bundle, namespace string, ope
 func CryostatRecordingTest(bundle *apimanifests.Bundle, namespace string, openShiftCertManager bool) (result scapiv1alpha3.TestResult) {
 	tr := newTestResources(CryostatRecordingTestName)
 	r := tr.TestResult
+	ch := make(chan *ContainerLog)
 
 	err := setupCRTestResources(tr, openShiftCertManager)
 	if err != nil {
 		return fail(*r, fmt.Sprintf("failed to set up %s test: %s", CryostatRecordingTestName, err.Error()))
 	}
+	defer cleanupAndLogs(&result, tr.Client, CryostatRecordingTestName, namespace, &ch)
 
 	// Create a default Cryostat CR
 	cr, err := createAndWaitTillCryostatAvailable(newCryostatCR(CryostatRecordingTestName, namespace, !tr.OpenShift), tr)
 	if err != nil {
 		return fail(*r, fmt.Sprintf("failed to determine application URL: %s", err.Error()))
 	}
-	ch, err := StartLogs(tr.Client.Clientset, cr)
+	ch, err = StartLogs(tr.Client.Clientset, cr)
 	if err != nil {
-		return fail(*r, fmt.Sprintf("failed to retrieve logs for the application: %s", err.Error()))
+		r.Log += fmt.Sprintf("failed to retrieve logs for the application: %s", err.Error())
 	}
```

Comment:
The channel returned from `StartLogs` should be used to collect logs instead? Maybe call `defer cleanupAndLogs` after creating CR?

Reply:
the point of all this refractoring is to get cleanupAndLogs right after the first return fail. The channel returned from StartLogs is used to collect the logs- that's why I pass in the pointer `*chan *ContainerLog` through `cleanupAndLogs` to answer your comment above^. This way, cleanupAndLogs has access to the updated channel after collecting the logs.

Reply:
It took a while for me to figure out a solution that allows `cleanupAndLogs` to be called before creating Cryostat CR and allows both workload and container logs to work :cry: please let me know if you have a better alternative

Reply:
you should still be able to see the container logs if you run :)

Reply:
Ahh I see. No I think it reasonable to me. Maybe, just declare the channel as below so it does not look like its using the initial one.


```go
var ch chan *ContainerLog
```

Reply:
Otherwise, we can attach and initialize the channel to the `TestResources` and pass it in the deferred cleanup func. WDUT?